exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/spill/RecordBatchSizer.java

```diff
@@ -129,11 +143,16 @@ public ColumnSize(ValueVector v, String prefix) {
         // No standard size for Union type
         dataSize = v.getPayloadByteCount(valueCount);
         break;
+      case GENERIC_OBJECT:
+        // We cannot provide a size for Generic Objects
```

Comment:
Execution gets here in some of the HashAgg functional tests. Probably in the case where varchars are aggregated, since as you explained to me varchars are stored in object vectors on heap.

Reply:
The `GENERIC_OBJECT` type is used any time we do a system table query: system tables are represented as Java objects.

There is an open question about certain aggregate functions. (See a note sent a week or so ago.) These aggregate functions use an `ObjectHolder` as their `\@Workspace`. @Ben-Zvi and I discussed whether such aggregates are spillable. This may be an unresolved issue.